timesketch/lib/llms/features/llm_forensic_report.py

```diff
@@ -0,0 +1,304 @@
+        unique_df = events_df.drop_duplicates(subset="combined_key", keep="first")
+
+        events_dict = (
+            unique_df[["datetime_str", "message"]]
+            .rename(columns={"datetime_str": "datetime"})
+            .to_dict(orient="records")
+        )
+
+        total_events_count = len(events_df)
+        unique_events_count = len(unique_df)
+
+        METRICS["llm_forensic_report_events_processed_total"].labels(
+            sketch_id=str(sketch.id)
+        ).inc(total_events_count)
+
+        METRICS["llm_forensic_report_unique_events_total"].labels(
+            sketch_id=str(sketch.id)
+        ).inc(unique_events_count)
+
+        if not events_dict:
+            return "No events to analyze for forensic report."
+
+        return self._get_prompt_text(events_dict)
+
+    def process_response(self, llm_response: Any, **kwargs: Any) -> dict[str, Any]:
+        """Processes the LLM response and creates a Story in the sketch.
+        Args:
+            llm_response: The response from the LLM model, expected to be a dictionary.
+            **kwargs: Additional arguments including:
+                - sketch_id: ID of the sketch being processed.
+                - sketch: The Sketch object.
+                - form: Form data containing query and filter information.
+                - datastore: OpenSearch instance for querying.
+                - timeline_ids: List of timeline IDs to query.
+        Returns:
+            Dictionary containing the processed response:
+                - summary: The forensic report text
+                - summary_event_count: Total number of events analyzed
+                - summary_unique_event_count: Number of unique events analyzed
+                - story_id: ID of the created story
+        Raises:
+            ValueError: If required parameters are missing or if the LLM response
+                      is not in the expected format.
+        """
+        sketch = kwargs.get("sketch")
+        form = kwargs.get("form")
+        datastore = kwargs.get("datastore")
+        timeline_ids = kwargs.get("timeline_ids")
+
+        if not sketch:
+            raise ValueError("Missing 'sketch' in kwargs")
+
+        if not form:
+            raise ValueError("Missing 'form' data in kwargs")
+
+        if not isinstance(llm_response, dict):
+            raise ValueError("LLM response is expected to be a dictionary")
+
+        summary_text = llm_response.get("summary")
+        if summary_text is None:
+            raise ValueError("LLM response missing 'summary' key")
+
+        query_filter = form.get("filter", {})
+        query_string = form.get("query", "*") or "*"
+
+        events_df = self._run_timesketch_query(
+            sketch,
+            query_string,
+            query_filter,
+            datastore=datastore,
+            timeline_ids=timeline_ids,
+        )
+
+        total_events_count = len(events_df)
```

Comment:
Unlikely since the icon only shows when there are stared events, but maybe catch if there are no results.